### ISSUE TITLE
EvolveGCN

### ISSUE THREAD

**Issue by johnny12150** [rusty1s/pytorch_geometric#1962]
## ❓ Questions & Help

Is there a future plan to implement more evolving graph models such as EvolveGCN?

I have noticed the [pytorch_geometric_temporal](https://github.com/benedekrozemberczki/pytorch_geometric_temporal) has implemented the model, but I am not sure how to feed the temporal dataset into it.

If I want to implement the EvolveGCN model on PyG, is there any advice to process the custom dataset?

**Comment by rusty1s** [comment]
We have plans to integrate more and more temporal graph models into PyG, and we just added the [Temporal Graph Networks](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.models.TGNMemory) model. Thanks for the pointer, I will definitely consider it to add to PyG. If you have any problems with PyTorch Geometric Temporal, I suggest to create an issue there to ask for help.

**Comment by johnny12150** [comment]
If I want to add some custom datasets for the TGN model, how should I do it?

**Comment by rusty1s** [comment]
We just added a `TemporalData` object, which defines an event stream of `src` and `dst` nodes, a timestamp `t`, and additional information (such as message features`msg` or labels `y`).

**Comment by johnny12150** [comment]
This `TemporalData`object is designed for temporal graph models such as the `TGN`?

**Comment by rusty1s** [comment]
Yes, this data format expects a list of temporal events, while EvolveGCN expects a list of temporal graphs. In general, you can make use of `torch_geometric.data.Data` to implement such a model (where each timestamp is denoted by a single data object), e.g.:
```python
for i in range(0, len(dataset), batch_size):
    optimizer.zero_grad():
        for j in range(batch_size):
             out = model(dataset[i + j])
             # perform some recurrent ops such as LSTMs
    # Loss computation
    optimizer.step()
```
